pipeline-basic/README.md

```diff
@@ -0,0 +1,145 @@
+
+* Deploying an integrated Jenkins server inside of OpenShift
+* Running both custom and oob Jenkins slaves as pods in OpenShift
+* "One Click" instantiation of a Jenkins Pipeline using OpenShift's Jenkins Pipeline Strategy feature
+* Promotion of an application's container image within an OpenShift Cluster (using `oc tag`)
+* Promotion of an application's container image to a separate OpenShift Cluster (using `skopeo`)
+
+## Architecture
+
+### OpenShift Templates
+
+The components of this pipeline are divided into two templates.
+
+The first, `generic-java-jenkins-pipeline` contains:
+
+* A `jenkinsPipelineStrategy` BuildConfig
+* An `s2i` BuildConfig
+* An ImageStream for the s2i build config to push to
+
+The second, `jws30-tomcat8-deployment` contains:
+
+* A tomcat8 DeploymentConfig
+* A Service definition
+* A Route
+
+The idea behind the split between the templates is that I can deploy the build template only once (to my dev project) and that the pipeline will promote my image through all of the various stages of my application's lifecycle. The deployment template gets deployed once to each of the stages of the application lifecycle (once per OpenShift project).
+
+### Pipeline Scipt
+
+This project includes a sample `pipeline.groovy` Jenkins Pipeline script that could be included with a Java project in order to implement a basic CI/CD pipeline for that project, under the following assumptions:
+
+* The project is build with Maven
+* The `pipeline.groovy` script is placed in the same directory as the `pom.xml` file in the git source.
+* The OpenShift projects that represent the Application's lifecycle stages are of the naming format: `<app-name>-dev`, `<app-name>-stage`, `<app-name>-prod`.
+
+For convenience, this pipeline script is already included in the following git repository, based on the [JBoss Developers Ticket Monster](https://github.com/jboss-developer/ticket-monster) app.
+
+https://github.com/etsauer/ticket-monster
+
+## Bill of Materials
+
+* One or Two OpenShift Container Platform Clusters
+  * Either OCP 3.4, or OCP 3.3 with the Pipelines tech preview feature enabled.
+* Access to GitHub
+
+## Implementation Instructions
+
+### 0. Pre-Work
+
+This pipeline has some dependencies that we need to provide it in our OpenShift cluster in order to work properly. They are:
+
+* The `jenkins-slave-image-mgmt` image from this repository
+* A slightly customized `jenkins2` image
+
+Both images can be built and provided to your cluster as follows:
+
+```
+oc process -f jenkins-slaves/templates/jenkins-slave-image-mgmt-template.json | oc create -f - -n openshift
+oc process -f cicd/jenkins-s2i/jenkins-s2i.yml -v JENKINS_GIT_URL=https://github.com/redhat-cop/containers-quickstarts.git -v JENKINS_GIT_CONTEXT_DIR=cicd/jenkins-s2i | oc create -f - -n openshift
```

Comment:
This is no longer needed as the kubernetes-plugin is included in the latest Jenkins image